compiler/src/language_server/sourcetree.re

```diff
@@ -137,11 +137,11 @@ module Make = (Ord: OrderableSegment) => {
 module type Sourcetree = {
   include SegmentTree with type point = Protocol.position;
   type node =
-    | Expression(Typedtree.expression, option(Types.value_description))
+    | Value(Env.t, Types.type_expr, Location.t)
     | Type(Typedtree.core_type)
     | Pattern(Typedtree.pattern)
```

Comment:
Why don't these also have `Location.t`?

Reply:
Patterns have a location on `pat_loc`, so we don't need to duplicate it. Since I changed `Expression` to `Value`, it doesn't have a location property on it so I just added it to the variant.